### ISSUE TITLE
add dark mode =)

### ISSUE THREAD

**Issue by t0mmyz7** [thingsboard/thingsboard#3762]
Hi Guys

Having a light/dark theme option for the UI can be a great addition.

Cheers

**Comment by lucilenikkels** [comment]
Hi! Is this issue still open? I am wondering because I noticed there are some dark mode files in the repo.

If so, I would be interested in picking this up with a group of fellow students in Software Architecture 👍

**Comment by t0mmyz7** [comment]
that would be awesome! I haven't seen Dark Mode in the UI yet.

maybe the team can comment if it is planned to be released.

**Comment by lucilenikkels** [comment]
@t0mmyz7 Yes, I would like to hear about that. I'm not sure who to @ about this. Perhaps @ashvayka ?

**Comment by lucilenikkels** [comment]
@ashvayka Okay thank you for replying. We will look into this and see what we can manage in time!

**Comment by t0mmyz7** [comment]
@lucilenikkels @ashvayka 
I actually think that Dark mode is somewhat important :) 

But another more important UI/UX feature IMO is adding tool-tip to the timeseries graphs in mobile view

https://github.com/thingsboard/thingsboard/issues/4048

Cheers :)

**Comment by lucilenikkels** [comment]
@t0mmyz7 Ah that one was somewhere on the list for us as well actually :)

We will explore these options and get back to you when we have made progress

**Comment by TTvanWillegen** [comment]
So I've been looking into this, and it seems that dark mode is already somewhat implemented. 
If we take a look at https://github.com/thingsboard/thingsboard/blob/master/ui-ngx/src/theme.scss#L142-L144, we can see that some sort of dark mode has been implemented.

We can trigger this dark mode by changing the class of the `body` node from `tb-default` to `tb-dark`.
This changes the colour scheme from
![image](https://user-images.githubusercontent.com/15815200/112717350-58e8a600-8eec-11eb-972a-58a1dca885ab.png)
to
![image](https://user-images.githubusercontent.com/15815200/112717357-656cfe80-8eec-11eb-8504-b3b296766c2b.png)

What I couldn't find yet, but will look into, if if there is a button to switch between these themes.

Also, I don't really know what people feel like with the current dark colour scheme. This is easily adapted too, though, by changing https://github.com/thingsboard/thingsboard/blob/master/ui-ngx/src/theme.scss#L113-129

**Comment by t0mmyz7** [comment]
@TTvanWillegen good find, I guess the team started working on it at one point

indeed the colors for it need a touch, i would say more like this:

![image](https://user-images.githubusercontent.com/31729091/112806709-c24cee00-907f-11eb-93a8-bd0ceead2953.png)

![image](https://user-images.githubusercontent.com/31729091/112806754-ce38b000-907f-11eb-880d-ff2c7d2ac676.png)



and for sure a button somewhere to switch, maybe in the user profile menu, 

![image](https://user-images.githubusercontent.com/31729091/112806632-acd7c400-907f-11eb-872b-44955ff745c4.png)

**Comment by t0mmyz7** [comment]
@TTvanWillegen @lucilenikkels hey check out this feature request, I think many will appreciate as well

https://github.com/thingsboard/thingsboard/issues/3418

**Comment by TTvanWillegen** [comment]
I have created a PR to display the dark mode, if this is the preference as specified by the user in it's browser.
See #4361

**Comment by mykhailokornieiev** [comment]
Hi. Thank you for the suggestion. Such a feature request was previously added to our backlog for developers. Please follow our [Release Notes](https://thingsboard.io/docs/reference/releases/) to see all new additions and fixes to the platform in subsequent releases.

**Comment by mykhailokornieiev** [comment]
Hi. Thank you for the suggestion. Such a feature request was previously added to our backlog for developers. While the feature request is valid, there are no plans to implement it in the nearest releases. Due to it, we've created an Idea discussion by the [link](https://github.com/thingsboard/thingsboard/discussions/14741) so that we can check the community's opinion on the necessity of reviewing the priority of implementing this feature request.
